beamer/tests/agent/multiple_processors/test_multiple_event_processors.py

```diff
@@ -0,0 +1,241 @@
+from beamer.agent.util import TokenChecker, make_web3, transact
+from beamer.tests.util import Sleeper
+
+SLAVE_TEST_PATH = Path(__file__).parent / "test_slave.py"
+CONFIG_PATH = os.path.join(os.getcwd(), "ape-config.yaml")
+BACKUP_PATH = os.path.join(os.getcwd(), "ape-config.yaml.2")
+
+
+# ape local account, to be used for fulfilling requests.
+# The private key here corresponds to the 10th account ganache creates on
+# startup.
+_LOCAL_ACCOUNT = ape.accounts.test_accounts[-1]
+
+
+def _create_and_return_backup_config() -> dict[str, Any]:
+    with open(CONFIG_PATH) as _f:
+        current_config = yaml.load(_f, Loader=SafeLoader)
+    with open(BACKUP_PATH, "w") as _f:
+        yaml.dump(current_config, _f, sort_keys=False, default_flow_style=False)
+    return current_config
+
+
+def _create_new_config(port: int):
+    current_config = _create_and_return_backup_config()
+    current_config["ganache"]["server"]["port"] = port
+    with open(CONFIG_PATH, "w") as _f:
+        yaml.dump(current_config, _f, sort_keys=False, default_flow_style=False)
+
+
+def _restore_backup_config(config: dict[str, Any]):
+    os.remove(CONFIG_PATH)
+    os.remove(BACKUP_PATH)
+    with open(CONFIG_PATH, "w") as _f:
+        yaml.dump(config, _f, sort_keys=False, default_flow_style=False)
+
+
+def _start_ganache_process(chain_id: ChainId, port: str) -> psutil.Popen:
+    cmd = [
+        "ganache-cli",
+        "--chain.vmErrorsOnRPCResponse",
+        "true",
+        "--server.port",
+        str(port),
+        "--miner.blockGasLimit",
+        "12000000",
+        "--wallet.totalAccounts",
+        "10",
+        "--hardfork",
+        "istanbul",
+        "--wallet.mnemonic",
+        "brownie",
+        "--chain.chainId",
+        str(chain_id),
+    ]
+    return psutil.Popen(cmd, stdin=DEVNULL, stdout=PIPE, stderr=PIPE)
+
+
+def _start_ganache(chain_id: ChainId, port: str) -> psutil.Popen:
+    return _start_ganache_process(chain_id, port)
+
+
+def _kill_ganache(process: psutil.Popen):
+    os.kill(process.pid, signal.SIGKILL)
+    process.wait()
+
+
+def _start_slave_test(port: int, request_count: int) -> psutil.Popen:
+    _create_new_config(port)
+    return psutil.Popen(
+        ["ape", "test", str(SLAVE_TEST_PATH), "-s"],
+        bufsize=0,
+        stdin=PIPE,
+        stdout=PIPE,
+        stderr=PIPE,
+        env=dict(os.environ, PORT=str(port), REQUEST_COUNT=str(request_count)),
+    )
+
+
+@contextlib.contextmanager
+def _new_networks(chain_port_map):
+    processes = {}
+    try:
+        current_config = _create_and_return_backup_config()
+        for chain_id, port in chain_port_map.items():
+            processes[chain_id] = _start_ganache(chain_id, port)
+        yield current_config
+    finally:
+        for process in processes.values():
+            _kill_ganache(process)
+
+
+def _get_chain_map() -> dict[ChainId, int]:
+    number_of_chains = 4
+    return {ChainId(i): 8545 + i for i in range(1, number_of_chains + 1)}
```

Comment:
I would avoid using low chain IDs, especially chain ID 1.

Why not just use port numbers as chain IDs?
```python
    return {ChainId(i): i for i in range(8546, 8546 + number_of_chains)}
```